internal/bootstrap/bootstrap_config_test.go

```diff
@@ -1540,3 +1591,238 @@ func TestConsulTagSpecifiers(t *testing.T) {
 		})
 	}
 }
+
+func TestAppendHCPMetrics(t *testing.T){
+	tests := map[string]struct{
+		inputArgs *BootstrapTplArgs
+		bindSocketDir string
+		wantArgs *BootstrapTplArgs
+	}{
+		"dir-without-trailing-slash":{
+			inputArgs: &BootstrapTplArgs{
+				Namespace: "default",
+				ProxyID: "web-sidecar-proxy",
+			},
+			bindSocketDir: "/tmp/consul/hcp-metrics",
+			wantArgs: &BootstrapTplArgs{
+				Namespace: "default",
+				ProxyID: "web-sidecar-proxy",
+				StatsSinksJSON: `{
+					"name": "envoy.stat_sinks.metrics_service",
```

Comment:
If this is repeated in a bunch of locations, maybe it's best to define it as a constant in the test somewhere.

Reply:
Yeah we weren't sure if inline is easier to read (gotta jump around) but I refactored it since it's a lot ! Thanks :)